### PR TITLE
Holographic remoting support for mixed reality camera manager

### PR DESCRIPTION
**Pull request by rolandsmeenk**
Overview
---
Initialization of transparent display settings when holographic remoting in editor using `HolographicEmulationWindow  emulationMode` to detect if remoting to HoloLens in Editor.

Changes
---
- Fixes: #1287

### CODE REVIEW

File: Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs [hunk 1]
```diff
@@ -62,29 +66,38 @@ public enum DisplayType
...
+            // Editor Hololens Remoting detection
+            var window = (HolographicEmulationWindow)EditorWindow.GetWindow(typeof(HolographicEmulationWindow));
+            if (window != null && window.emulationMode == EmulationMode.RemoteDevice)
+            {
+                CurrentDisplayType = DisplayType.Transparent;
+            }
+#endif
+
+            if (CurrentDisplayType == DisplayType.Opaque)
+                ApplySettingsForOpaqueDisplay();
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs]: Nit: braces

File: Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs [hunk 2]
```diff
@@ -62,29 +66,38 @@ public enum DisplayType
...
+            else
+                ApplySettingsForTransparentDisplay();
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs]: Nit: braces

**Comment by rolandsmeenk** [comment]
Old company coding guideline habits, but I am fine with the proposed changes. Where do I accept them? Or do I have to push those changes to my branch?

**Comment by StephenHodgson** [comment]
Push the changes to your branch.

File: Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs [hunk 3]
```diff
@@ -62,29 +66,38 @@ public enum DisplayType
...
-                    return;
                 }
 #endif
             }
 
-            CurrentDisplayType = DisplayType.Opaque;
-            ApplySettingsForOpaqueDisplay();
+#if UNITY_EDITOR
+            // Editor Hololens Remoting detection
+            var window = (HolographicEmulationWindow)EditorWindow.GetWindow(typeof(HolographicEmulationWindow));
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs]: You wouldn't have to cast here if explicit var declaration

File: Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs [hunk 4]
```diff
@@ -62,29 +66,42 @@ public enum DisplayType
...
-                    return;
                 }
 #endif
             }
 
-            CurrentDisplayType = DisplayType.Opaque;
-            ApplySettingsForOpaqueDisplay();
+#if UNITY_EDITOR
+            // Editor Hololens Remoting detection
+            var window = EditorWindow.GetWindow<HolographicEmulationWindow>(typeof(HolographicEmulationWindow));
```
Comment (StephenHodgson) [inline Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs]: If users don't have the `HolographicEmulationWindow` open in the editor we never get a valid window.

Comment (rolandsmeenk) [reply Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs]: Hmm, just tested and the behavior is different. GetWindow creates a new window if there is no window open and it defaults to emulationmode.none.

Comment (StephenHodgson) [reply Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs]: That's what I'm saying.

Comment (rolandsmeenk) [reply Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs]: Ok, so now we wait for Unity to expose something that may be more helpful?

Comment (StephenHodgson) [reply Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs]: That would be nice, but I suppose this works in the meantime.

I'd just add a note in the comments or something.

**Comment by keveleigh** [comment]
I'm working on getting official word if we're working around a Unity bug here or if this is by-design. It feels like `IsDisplayOpaque` should return false if it knows we're remoting to a HoloLens.

**Comment by NeerajW** [comment]
Kurtis is following up with a bug id: Ideally when remoting to HoloLens you should get IsDisplayOpaque as false.

**Comment by StephenHodgson** [comment]
Where's the bug report?

**Comment by keveleigh** [comment]
Interesting. After further investigating, this isn't a bug in `IsDisplayOpaque`, but a bug in this script. 

Instead of all the checks for remoting, [the `Application.IsEditor` check](https://github.com/rolandsmeenk/MixedRealityToolkit-Unity/blob/673831907adab3e27eebc0883a15a2640cdeeb36/Assets/HoloToolkit/Input/Scripts/Utilities/Managers/MixedRealityCameraManager.cs#L71) should be removed. The APIs work properly no matter where they're called from, so we should let them handle it.

**Comment by keveleigh** [comment]
I added a PR into this PR that should resolve the issue without needed to detect specific Editor windows.

**Comment by keveleigh** [comment]
@rolandsmeenk Please take a look at https://github.com/rolandsmeenk/MixedRealityToolkit-Unity/pull/1 when you get a chance, and pull it into here if it still fixes the issues you were seeing!

**Comment by rolandsmeenk** [comment]
Hi Keve, did not realize you were waiting for me. This tested fine when remoting to HL so merged it.

**Comment by keveleigh** [comment]
Thanks @rolandsmeenk!

**Pull request merged**